react/features/screenshot-capture/ScreenshotCaptureSummary.js

```diff
@@ -208,8 +220,13 @@ export default class ScreenshotCaptureSummary {
             .setReturnEarlyThreshold(PERCENTAGE_LOWER_BOUND)
             .onComplete(resultData => {
                 if (resultData.rawMisMatchPercentage > PERCENTAGE_LOWER_BOUND) {
-                    this._doProcessScreenshot(imageData);
+                    try {
+                        this._doProcessScreenshot(imageData);
+                    } catch (err) {
+                        logger.error(err);
+                    }
                 }
             });
+        this._queue.splice(this._queue.indexOf(timestamp), 1);
```

Comment:
shouldn't this happen inside onComplete? otherwise you'll still run into the problem because it's `_doProcessScreenshot` what may take long. In fact, `_doProcessScreenshot` should be async and you should be awaiting before clearing the item from the queue, otherwise you may end up with an unbound number of concurrent fetch() requests.